src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -834,6 +1089,77 @@ private void AddMetaKeyValues(int i, DataViewSchema.Annotations.Builder builder)
             }
 
             protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer) => _parent.GetGetterCore(input, iinfo, out disposer);
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariable, string dstVariable)
+            {
+                string opType;
+                OnnxNode murmurNode;
+
+                opType = "MurmurHash3";
+                string murmurOutput = (_srcTypes[iinfo].RawType == typeof(ReadOnlyMemory<char>)) ? dstVariable: ctx.AddIntermediateVariable(_dstTypes[iinfo], "MurmurOutput");
+                var srcType = _srcTypes[iinfo].GetItemType().RawType;
+
+                // Numeric input types are limited to those supported by the Onnxruntime MurmurHash operator, which currently only supports
+                // uints and ints. Thus, ulongs, longs, doubles, floats, and booleans are not supported.
+                if (srcType == typeof(ushort) || srcType == typeof(short) || srcType == typeof(sbyte) || srcType == typeof(byte))
```

Comment:
there is no support for Key types (if (_types[iinfo].RawType == typeof(KeyDataViewType))), key types need to have +1 at the end of the hash value.

Reply:
For the unsigned integer types, you need to look at `srcTypes[iinfo].GetItemType()` itself, to see if it is a key type or not.

---
In reply to: [407106422](https://github.com/dotnet/machinelearning/pull/5013#discussion_r407106422) [](ancestors = 407106422)

Reply:
Added support for keytype back.